Compile of 0.23.15 with libfmt-11.0.0 fails

<!-- See https://www.musicpd.org/help/ -->
## Bug report
### Describe the bug
Minor api change in libfmt-11.0.0 causes mod to fail compilation
Compiles successfully with libfmt 10.2.1

## Expected Behavior
Successful compile
I believe the error is the same one solved in https://github.com/MariaDB/server/pull/3276/commits/5ece30a9fac3ff4cba26435c7d559458aebeae48

## Actual Behavior
Failed compile

## Version
<!-- Paste the output of "mpd --version" here -->
0.23.15

## Configuration
<!-- Paste your MPD configuration here -->
Na

## Log
<!-- Paste relevant portions of the log file here (--verbose) -->
<!-- If MPD has crashed, include a backtrace (see https://mpd.readthedocs.io/en/stable/user.html#mpd-crashes) -->
```
Cleaning... 0 files.
[87/600] Compiling C++ object liblog.a.p/src_Log.cxx.o
FAILED: liblog.a.p/src_Log.cxx.o 
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -Iliblog.a.p -I. -I.. -Isrc -I../src -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -ffast-math -ftree-vectorize -Wcast-qual -Wdouble-promotion -Wmissing-declarations -Wshadow -Wunused -Wvla -Wwrite-strings -Wno-stringop-overflow -fno-threadsafe-statics -fmerge-all-constants -Wcomma-subscript -Wextra-semi -Wmismatched-tags -Woverloaded-virtual -Wsign-promo -Wvolatile -Wvirtual-inheritance -Wno-non-virtual-dtor -Wsuggest-override -fvisibility=hidden -ffunction-sections -fdata-sections -D_GNU_SOURCE -march=x86-64-v3 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -fPIC -MD -MQ liblog.a.p/src_Log.cxx.o -MF liblog.a.p/src_Log.cxx.o.d -o liblog.a.p/src_Log.cxx.o -c ../src/Log.cxx
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/format.h:41,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/core.h:5,
                 from ../src/Log.hxx:25,
                 from ../src/Log.cxx:20:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h: In instantiation of 'static void fmt::v10::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = std::__exception_ptr::exception_ptr; Formatter = fmt::v10::formatter<std::__exception_ptr::exception_ptr>; Context = fmt::v10::context; typename Context::parse_context_type = fmt::v10::basic_format_parse_context<char>]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1373:19:   required from 'fmt::v10::detail::value<Context>::value(T&) [with T = const std::__exception_ptr::exception_ptr; Context = fmt::v10::context]'
 1373 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1374 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1631:41:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::context; T = const std::__exception_ptr::exception_ptr; typename std::enable_if<PACKED, int>::type <anonymous> = 0]'
 1631 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:2002:74:   required from 'constexpr fmt::v10::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v10::make_format_args(T& ...) [with Context = context; T = {const char*, const std::__exception_ptr::exception_ptr}; long unsigned int NUM_ARGS = 2; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 252; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]'
 2002 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2003 |       args)...}};
      |       ~~~~~                                                               
../src/Log.hxx:50:31:   required from 'void LogFmt(LogLevel, const Domain&, const S&, Args&& ...) [with S = char [7]; Args = {const char*&, const std::__exception_ptr::exception_ptr&}]'
   50 |                        fmt::make_format_args(args...));
      |                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
../src/Log.cxx:50:8:   required from here
   50 |         LogFmt(level, exception_domain, "{}: {}", msg, ep);
      |         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1392:29: error: passing 'const fmt::v10::formatter<std::__exception_ptr::exception_ptr>' as 'this' argument discards qualifiers [-fpermissive]
 1392 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/Log.cxx:21:
../src/lib/fmt/ExceptionFormatter.hxx:41:14: note:   in call to 'auto fmt::v10::formatter<std::__exception_ptr::exception_ptr>::format(std::__exception_ptr::exception_ptr, FormatContext&) [with FormatContext = fmt::v10::context]'
   41 |         auto format(std::exception_ptr e, FormatContext &ctx) {
      |              ^~~~~~
[92/600] Compiling C++ object src/event/libevent.a.p/ServerSocket.cxx.o
FAILED: src/event/libevent.a.p/ServerSocket.cxx.o 
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -Isrc/event/libevent.a.p -Isrc/event -I../src/event -Isrc -I../src -I. -I.. -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -ffast-math -ftree-vectorize -Wcast-qual -Wdouble-promotion -Wmissing-declarations -Wshadow -Wunused -Wvla -Wwrite-strings -Wno-stringop-overflow -fno-threadsafe-statics -fmerge-all-constants -Wcomma-subscript -Wextra-semi -Wmismatched-tags -Woverloaded-virtual -Wsign-promo -Wvolatile -Wvirtual-inheritance -Wno-non-virtual-dtor -Wsuggest-override -fvisibility=hidden -ffunction-sections -fdata-sections -D_GNU_SOURCE -march=x86-64-v3 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -fPIC -DBOOST_ALL_NO_LIB -MD -MQ src/event/libevent.a.p/ServerSocket.cxx.o -MF src/event/libevent.a.p/ServerSocket.cxx.o.d -o src/event/libevent.a.p/ServerSocket.cxx.o -c ../src/event/ServerSocket.cxx
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/format.h:41,
                 from ../src/lib/fmt/ExceptionFormatter.hxx:35,
                 from ../src/event/ServerSocket.cxx:22:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h: In instantiation of 'static void fmt::v10::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = std::__exception_ptr::exception_ptr; Formatter = fmt::v10::formatter<std::__exception_ptr::exception_ptr>; Context = fmt::v10::context; typename Context::parse_context_type = fmt::v10::basic_format_parse_context<char>]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1373:19:   required from 'fmt::v10::detail::value<Context>::value(T&) [with T = std::__exception_ptr::exception_ptr; Context = fmt::v10::context]'
 1373 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1374 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1631:41:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::context; T = std::__exception_ptr::exception_ptr; typename std::enable_if<PACKED, int>::type <anonymous> = 0]'
 1631 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:2002:74:   required from 'constexpr fmt::v10::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v10::make_format_args(T& ...) [with Context = context; T = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__exception_ptr::exception_ptr}; long unsigned int NUM_ARGS = 3; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 4061; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]'
 2002 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2003 |       args)...}};
      |       ~~~~~                                                               
../src/Log.hxx:50:31:   required from 'void LogFmt(LogLevel, const Domain&, const S&, Args&& ...) [with S = char [79]; Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__exception_ptr::exception_ptr&}]'
   50 |                        fmt::make_format_args(args...));
      |                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
../src/Log.hxx:99:8:   required from 'void FmtError(const Domain&, const S&, Args&& ...) [with S = char [79]; Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__exception_ptr::exception_ptr}]'
   99 |         LogFmt(LogLevel::ERROR, domain, format_str, args...);
      |         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/event/ServerSocket.cxx:231:13:   required from here
  231 |                                 FmtError(server_socket_domain,
      |                                 ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
  232 |                                          "bind to '{}' failed "
      |                                          ~~~~~~~~~~~~~~~~~~~~~~
  233 |                                          "(continuing anyway, because "
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  234 |                                          "binding to '{}' succeeded): {}",
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  235 |                                          address_string,
      |                                          ~~~~~~~~~~~~~~~
  236 |                                          good_string,
      |                                          ~~~~~~~~~~~~
  237 |                                          std::current_exception());
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1392:29: error: passing 'const fmt::v10::formatter<std::__exception_ptr::exception_ptr>' as 'this' argument discards qualifiers [-fpermissive]
 1392 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/lib/fmt/ExceptionFormatter.hxx:41:14: note:   in call to 'auto fmt::v10::formatter<std::__exception_ptr::exception_ptr>::format(std::__exception_ptr::exception_ptr, FormatContext&) [with FormatContext = fmt::v10::context]'
   41 |         auto format(std::exception_ptr e, FormatContext &ctx) {
      |              ^~~~~~
[94/600] Compiling C++ object src/lib/alsa/libalsa.a.p/HwSetup.cxx.o
FAILED: src/lib/alsa/libalsa.a.p/HwSetup.cxx.o 
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -Isrc/lib/alsa/libalsa.a.p -Isrc/lib/alsa -I../src/lib/alsa -Isrc -I../src -I. -I.. -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -ffast-math -ftree-vectorize -Wcast-qual -Wdouble-promotion -Wmissing-declarations -Wshadow -Wunused -Wvla -Wwrite-strings -Wno-stringop-overflow -fno-threadsafe-statics -fmerge-all-constants -Wcomma-subscript -Wextra-semi -Wmismatched-tags -Woverloaded-virtual -Wsign-promo -Wvolatile -Wvirtual-inheritance -Wno-non-virtual-dtor -Wsuggest-override -fvisibility=hidden -ffunction-sections -fdata-sections -D_GNU_SOURCE -march=x86-64-v3 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -fPIC -MD -MQ src/lib/alsa/libalsa.a.p/HwSetup.cxx.o -MF src/lib/alsa/libalsa.a.p/HwSetup.cxx.o.d -o src/lib/alsa/libalsa.a.p/HwSetup.cxx.o -c ../src/lib/alsa/HwSetup.cxx
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/format.h:41,
                 from ../src/lib/fmt/AudioFormatFormatter.hxx:36,
                 from ../src/lib/alsa/HwSetup.cxx:23:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h: In instantiation of 'static void fmt::v10::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = SampleFormat; Formatter = fmt::v10::formatter<SampleFormat>; Context = fmt::v10::context; typename Context::parse_context_type = fmt::v10::basic_format_parse_context<char>]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1373:19:   required from 'fmt::v10::detail::value<Context>::value(T&) [with T = SampleFormat; Context = fmt::v10::context]'
 1373 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1374 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1631:41:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::context; T = SampleFormat; typename std::enable_if<PACKED, int>::type <anonymous> = 0]'
 1631 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:2002:74:   required from 'constexpr fmt::v10::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v10::make_format_args(T& ...) [with Context = context; T = {SampleFormat}; long unsigned int NUM_ARGS = 1; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 15; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]'
 2002 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2003 |       args)...}};
      |       ~~~~~                                                               
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/format.h:4357:44:   required from 'std::string fmt::v10::format(format_string<T ...>, T&& ...) [with T = {SampleFormat&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, SampleFormat&>]'
 4357 |   return vformat(fmt, fmt::make_format_args(args...));
      |                       ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
../src/lib/alsa/HwSetup.cxx:201:22:   required from here
  201 |                                       fmt::format("Failed to configure format {}",
      |                                       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  202 |                                                   audio_format.format).c_str());
      |                                                   ~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1392:29: error: passing 'const fmt::v10::formatter<SampleFormat>' as 'this' argument discards qualifiers [-fpermissive]
 1392 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/lib/fmt/AudioFormatFormatter.hxx:42:14: note:   in call to 'auto fmt::v10::formatter<SampleFormat>::format(SampleFormat, FormatContext&) [with FormatContext = fmt::v10::context]'
   42 |         auto format(const SampleFormat format, FormatContext &ctx) {
      |              ^~~~~~
[96/600] Compiling C++ object src/event/libevent.a.p/Thread.cxx.o
FAILED: src/event/libevent.a.p/Thread.cxx.o 
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -Isrc/event/libevent.a.p -Isrc/event -I../src/event -Isrc -I../src -I. -I.. -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -ffast-math -ftree-vectorize -Wcast-qual -Wdouble-promotion -Wmissing-declarations -Wshadow -Wunused -Wvla -Wwrite-strings -Wno-stringop-overflow -fno-threadsafe-statics -fmerge-all-constants -Wcomma-subscript -Wextra-semi -Wmismatched-tags -Woverloaded-virtual -Wsign-promo -Wvolatile -Wvirtual-inheritance -Wno-non-virtual-dtor -Wsuggest-override -fvisibility=hidden -ffunction-sections -fdata-sections -D_GNU_SOURCE -march=x86-64-v3 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -fPIC -DBOOST_ALL_NO_LIB -MD -MQ src/event/libevent.a.p/Thread.cxx.o -MF src/event/libevent.a.p/Thread.cxx.o.d -o src/event/libevent.a.p/Thread.cxx.o -c ../src/event/Thread.cxx
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/format.h:41,
                 from ../src/lib/fmt/ExceptionFormatter.hxx:35,
                 from ../src/event/Thread.cxx:24:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h: In instantiation of 'static void fmt::v10::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = std::__exception_ptr::exception_ptr; Formatter = fmt::v10::formatter<std::__exception_ptr::exception_ptr>; Context = fmt::v10::context; typename Context::parse_context_type = fmt::v10::basic_format_parse_context<char>]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1373:19:   required from 'fmt::v10::detail::value<Context>::value(T&) [with T = std::__exception_ptr::exception_ptr; Context = fmt::v10::context]'
 1373 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1374 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1631:41:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::context; T = std::__exception_ptr::exception_ptr; typename std::enable_if<PACKED, int>::type <anonymous> = 0]'
 1631 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:2002:74:   required from 'constexpr fmt::v10::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v10::make_format_args(T& ...) [with Context = context; T = {std::__exception_ptr::exception_ptr}; long unsigned int NUM_ARGS = 1; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 15; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]'
 2002 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2003 |       args)...}};
      |       ~~~~~                                                               
../src/Log.hxx:50:31:   required from 'void LogFmt(LogLevel, const Domain&, const S&, Args&& ...) [with S = char [68]; Args = {std::__exception_ptr::exception_ptr&}]'
   50 |                        fmt::make_format_args(args...));
      |                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
../src/Log.hxx:75:8:   required from 'void FmtInfo(const Domain&, const S&, Args&& ...) [with S = char [68]; Args = {std::__exception_ptr::exception_ptr}]'
   75 |         LogFmt(LogLevel::INFO, domain, format_str, args...);
      |         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/event/Thread.cxx:64:11:   required from here
   64 |                         FmtInfo(event_domain,
      |                         ~~~~~~~^~~~~~~~~~~~~~
   65 |                                 "RTIOThread could not get realtime scheduling, continuing anyway: {}",
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   66 |                                 std::current_exception());
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1392:29: error: passing 'const fmt::v10::formatter<std::__exception_ptr::exception_ptr>' as 'this' argument discards qualifiers [-fpermissive]
 1392 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/lib/fmt/ExceptionFormatter.hxx:41:14: note:   in call to 'auto fmt::v10::formatter<std::__exception_ptr::exception_ptr>::format(std::__exception_ptr::exception_ptr, FormatContext&) [with FormatContext = fmt::v10::context]'
   41 |         auto format(std::exception_ptr e, FormatContext &ctx) {
      |              ^~~~~~
[102/600] Compiling C++ object src/lib/nfs/libnfs.a.p/Manager.cxx.o
FAILED: src/lib/nfs/libnfs.a.p/Manager.cxx.o 
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/bin/x86_64-libreelec-linux-gnu-g++ -Isrc/lib/nfs/libnfs.a.p -Isrc/lib/nfs -I../src/lib/nfs -Isrc -I../src -I. -I.. -I/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include -fdiagnostics-color=always -D_GLIBCXX_ASSERTIONS=1 -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Wpedantic -std=c++17 -ffast-math -ftree-vectorize -Wcast-qual -Wdouble-promotion -Wmissing-declarations -Wshadow -Wunused -Wvla -Wwrite-strings -Wno-stringop-overflow -fno-threadsafe-statics -fmerge-all-constants -Wcomma-subscript -Wextra-semi -Wmismatched-tags -Woverloaded-virtual -Wsign-promo -Wvolatile -Wvirtual-inheritance -Wno-non-virtual-dtor -Wsuggest-override -fvisibility=hidden -ffunction-sections -fdata-sections -D_GNU_SOURCE -march=x86-64-v3 -Wall -pipe -O2 -fomit-frame-pointer -DNDEBUG -fPIC -MD -MQ src/lib/nfs/libnfs.a.p/Manager.cxx.o -MF src/lib/nfs/libnfs.a.p/Manager.cxx.o.d -o src/lib/nfs/libnfs.a.p/Manager.cxx.o -c ../src/lib/nfs/Manager.cxx
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/format.h:41,
                 from ../src/lib/fmt/ExceptionFormatter.hxx:35,
                 from ../src/lib/nfs/Manager.cxx:21:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h: In instantiation of 'static void fmt::v10::detail::value<Context>::format_custom_arg(void*, typename Context::parse_context_type&, Context&) [with T = std::__exception_ptr::exception_ptr; Formatter = fmt::v10::formatter<std::__exception_ptr::exception_ptr>; Context = fmt::v10::context; typename Context::parse_context_type = fmt::v10::basic_format_parse_context<char>]':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1373:19:   required from 'fmt::v10::detail::value<Context>::value(T&) [with T = std::__exception_ptr::exception_ptr; Context = fmt::v10::context]'
 1373 |     custom.format = format_custom_arg<
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
 1374 |         value_type, typename Context::template formatter_type<value_type>>;
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1631:41:   required from 'constexpr fmt::v10::detail::value<Context> fmt::v10::detail::make_arg(T&) [with bool PACKED = true; Context = fmt::v10::context; T = std::__exception_ptr::exception_ptr; typename std::enable_if<PACKED, int>::type <anonymous> = 0]'
 1631 |   return {arg_mapper<Context>().map(val)};
      |                                         ^
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:2002:74:   required from 'constexpr fmt::v10::detail::format_arg_store<Context, NUM_ARGS, 0, DESC> fmt::v10::make_format_args(T& ...) [with Context = context; T = {const char*, const char*, std::__exception_ptr::exception_ptr}; long unsigned int NUM_ARGS = 3; long unsigned int NUM_NAMED_ARGS = 0; long long unsigned int DESC = 4044; typename std::enable_if<(NUM_NAMED_ARGS == 0), int>::type <anonymous> = 0]'
 2002 |   return {{detail::make_arg<NUM_ARGS <= detail::max_packed_args, Context>(
      |            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 2003 |       args)...}};
      |       ~~~~~                                                               
../src/Log.hxx:50:31:   required from 'void LogFmt(LogLevel, const Domain&, const S&, Args&& ...) [with S = char [25]; Args = {const char*&, const char*&, std::__exception_ptr::exception_ptr&}]'
   50 |                        fmt::make_format_args(args...));
      |                        ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
../src/Log.hxx:99:8:   required from 'void FmtError(const Domain&, const S&, Args&& ...) [with S = char [25]; Args = {const char*, const char*, std::__exception_ptr::exception_ptr&}]'
   99 |         LogFmt(LogLevel::ERROR, domain, format_str, args...);
      |         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/lib/nfs/Manager.cxx:34:10:   required from here
   34 |         FmtError(nfs_domain, "NFS error on '{}:{}': {}",
      |         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   35 |                  GetServer(), GetExportName(), e);
      |                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-13.0-devel/toolchain/x86_64-libreelec-linux-gnu/sysroot/usr/include/fmt/base.h:1392:29: error: passing 'const fmt::v10::formatter<std::__exception_ptr::exception_ptr>' as 'this' argument discards qualifiers [-fpermissive]
 1392 |     ctx.advance_to(cf.format(*static_cast<qualified_type*>(arg), ctx));
      |                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/lib/fmt/ExceptionFormatter.hxx:41:14: note:   in call to 'auto fmt::v10::formatter<std::__exception_ptr::exception_ptr>::format(std::__exception_ptr::exception_ptr, FormatContext&) [with FormatContext = fmt::v10::context]'
   41 |         auto format(std::exception_ptr e, FormatContext &ctx) {
      |              ^~~~~~
ninja: build stopped: subcommand failed.
```

--- a/src/lib/ffmpeg/LibFmt.hxx
+++ b/src/lib/ffmpeg/LibFmt.hxx
@@ -13,7 +13,7 @@ template<>
 struct fmt::formatter<AVSampleFormat> : formatter<string_view>
 {
 	template<typename FormatContext>
-	auto format(const AVSampleFormat format, FormatContext &ctx) {
+	auto format(const AVSampleFormat format, FormatContext &ctx) const {
 		const char *name = av_get_sample_fmt_name(format);
 		if (name == nullptr)
 			name = "?";

--- a/src/lib/fmt/AudioFormatFormatter.hxx
+++ b/src/lib/fmt/AudioFormatFormatter.hxx
@@ -13,7 +13,7 @@ template<>
 struct fmt::formatter<SampleFormat> : formatter<string_view>
 {
 	template<typename FormatContext>
-	auto format(const SampleFormat format, FormatContext &ctx) {
+	auto format(const SampleFormat format, FormatContext &ctx) const {
 		return formatter<string_view>::format(sample_format_to_string(format),
 						      ctx);
 	}
@@ -23,7 +23,7 @@ template<>
 struct fmt::formatter<AudioFormat> : formatter<string_view>
 {
 	template<typename FormatContext>
-	auto format(const AudioFormat &af, FormatContext &ctx) {
+	auto format(const AudioFormat &af, FormatContext &ctx) const {
 		return formatter<string_view>::format(ToString(af).c_str(),
 						      ctx);
 	}

--- a/src/lib/fmt/ExceptionFormatter.hxx
+++ b/src/lib/fmt/ExceptionFormatter.hxx
@@ -12,7 +12,7 @@ template<>
 struct fmt::formatter<std::exception_ptr> : formatter<string_view>
 {
 	template<typename FormatContext>
-	auto format(std::exception_ptr e, FormatContext &ctx) {
+	auto format(std::exception_ptr e, FormatContext &ctx) const {
 		return formatter<string_view>::format(GetFullMessage(e), ctx);
 	}
 };

--- a/src/lib/fmt/PathFormatter.hxx
+++ b/src/lib/fmt/PathFormatter.hxx
@@ -13,7 +13,7 @@ template<std::convertible_to<Path> T>
 struct fmt::formatter<T> : formatter<string_view>
 {
 	template<typename FormatContext>
-	auto format(Path path, FormatContext &ctx) {
+	auto format(Path path, FormatContext &ctx) const {
 		return formatter<string_view>::format(path.ToUTF8(), ctx);
 	}
 };

